src/lib/form-field/form-field.ts

```diff
@@ -86,6 +95,9 @@ let nextUniqueId = 0;
 export class MatFormField implements AfterViewInit, AfterContentInit, AfterContentChecked {
   private _labelOptions: LabelOptions;
 
+  /** The form-field style variant. */
+  @Input() variant: 'legacy' | 'standard' | 'box' = 'legacy';
+
```

Comment:
Sidenote: maybe we should add an injection token that allows you to specify the default variant?